### ISSUE TITLE
HP PRIMERA & 3PAR operating system and tooling versions (showcage)

### ISSUE THREAD

**Issue by petr-fischer** [qaxi/nagios-check_3par#8]
Hello!

on the HP PRIMERA arrays (showsys still prints "HPE_3PAR XXX" as model), there is newer OS version and different `showcage` command.

On our old 3PAR arrays, there is OS version (showversion) 3.3.2, but on PRIMERA, there is 4.5.0.

The new OS does not support the "-d" parameter for `showcage` command - so the *check_ps_cage* check does not work.

But on the other hand, `showcage` command now shows *State* column directly in the basic output, like this:

```bash
cli% showcage
Id Name  LoopA Pos.A LoopB Pos.B Drives Temp  RevA RevB Model FormFactor State
 0 cage0 0:0:4     0 1:0:4     0     20 28-29 0703 0703 DCN5  SFF        normal
 2 cage2 0:0:2     0 1:0:2     0     20 28-29 0703 0703 DCS12 SFF        normal
```

Some questions:

1) would it be good if this check can also do the HP PRIMERA (except for the "cage" everything seems to work)?

2) If we want to somehow branch (conditions) the check code for more HP array types, what is the best one? By OS version (`showversion`)? Any other trick?

Thanks for discussion and experiences with this HP arrays!

**Comment by qaxi** [comment]
Hello, 
1) Yes i making it suit HP Primera vould by fine.
2) OS version is allways good sign of OS capabilities ...

At the time i unfortunately have no HP 3PAR or Primera arrays to test on ... If you do have, would you continue the developement?

**Comment by petr-fischer** [comment]
Yes, we need PRIMERA support, so if you agree to include it, I will send a pull request in some time (maybe I'll finally learn awk).

**Comment by qaxi** [comment]
BTW: Please add at least README update ...
